pkg/controller/v1alpha2/applicationconfiguration/apply.go

```diff
@@ -131,6 +159,20 @@ func (a *workloads) dereferenceScope(ctx context.Context, namespace string, stat
 	return nil
 }
 
+// dereferenceAllScope dereferences workloads owned by the appConfig being deleted from the scopes they belong to.
+func (a *workloads) dereferenceAllScopes(ctx context.Context, namespace string, status []v1alpha2.WorkloadStatus) error {
+	for _, st := range status {
+		toBeDeferenced := st.Scopes
+		for _, s := range toBeDeferenced {
+			if err := a.applyScopeRemoval(ctx, namespace, st, s); err != nil {
```

Comment:
why pass in st again after it's already used here? Only the workloadRef from status is needed